### PR TITLE
feat: add external employess

### PR DESCRIPTION
**Pull request by trowik**
add new boolean property to employment model
disable statistics page for external employees
disable analysis page for external employees
show contract type in user's general information box

### CODE REVIEW

**Review by aziiee — APPROVED** [review]

File: app/abilities/page.js [hunk 1]
```diff
@@ -0,0 +1,6 @@
+import { not } from "@ember/object/computed";
+import { Ability } from "ember-can";
+
+export default Ability.extend({
```
Comment (anehx) [inline app/abilities/page.js:4]: Class syntax

File: app/analysis/edit/template.hbs [hunk 2]
```diff
@@ -1,3 +1,9 @@
+{{#if (cannot "access page")}}
+  <div class="text-center" data-test-access-forbidden>
+    <h1>Access forbidden</h1>
+    <h4>You do not have the permission to access this page</h4>
+  </div>
+{{else}}
 <div class="grid grid--12of12 grid-md--4of12">
```
Comment (anehx) [inline app/analysis/edit/template.hbs:7]: Indent incorrect

File: app/analysis/index/template.hbs [hunk 3]
```diff
@@ -1,6 +1,12 @@
 {{#if (or (media 'isMo') (media 'isXs') (media 'isSm'))}}
   {{no-mobile-message}}
 {{else}}
+  {{#if (cannot "access page")}}
+    <div class="text-center" data-test-access-forbidden>
+      <h1>Access forbidden</h1>
+      <h4>You do not have the permission to access this page</h4>
+    </div>
+  {{else}}
   <h1>Analysis</h1>
```
Comment (anehx) [inline app/analysis/index/template.hbs:10]: Indent incorrect

File: app/analysis/index/template.hbs [hunk 4]
```diff
@@ -1,6 +1,12 @@
 {{#if (or (media 'isMo') (media 'isXs') (media 'isSm'))}}
   {{no-mobile-message}}
 {{else}}
+  {{#if (cannot "access page")}}
+    <div class="text-center" data-test-access-forbidden>
+      <h1>Access forbidden</h1>
+      <h4>You do not have the permission to access this page</h4>
+    </div>
```
Comment (anehx) [inline app/analysis/index/template.hbs:8]: Make a component for this

File: app/components/sy-topnav/template.hbs [hunk 5]
```diff
@@ -20,21 +20,23 @@
...
+        </li>
+        <li class="nav-top-list-item">
+          {{#link-to 'statistics'}}
+            {{fa-icon 'bar-chart'}} Statistics
+          {{/link-to}}
+        </li>
+        <li class="nav-top-list-item">
+          {{#link-to 'projects'}}
+            {{fa-icon 'briefcase'}} Projects
+          {{/link-to}}
```
Comment (anehx) [inline app/components/sy-topnav/template.hbs:37]: Refactor to angle bracket syntax

File: app/index/template.hbs [hunk 6]
```diff
@@ -40,19 +40,21 @@
...
-          {{action (toggle 'showAddModal' this)}}
-        >
-          Add absence
-        </button>
+          <button
+            type="button"
+            data-test-add-absence
+            class="btn btn-default"
+            disabled={{cannot "access page"}}
+            {{action (toggle 'showAddModal' this)}}
```
Comment (anehx) [inline app/index/template.hbs:54]: `action` is deprecated, use `on`

File: app/index/template.hbs [hunk 7]
```diff
@@ -83,14 +85,15 @@
...
-            {{action (toggle 'showAddModal' this)}}
-          >
-            Add absence
-          </button>
+            <button
+              type="button"
+              data-test-add-absence
+              class="btn btn-default"
+              disabled={{cannot "access page"}}
+              {{action (toggle 'showAddModal' this)}}
```
Comment (anehx) [inline app/index/template.hbs:93]: Same here

File: app/models/employment.js [hunk 8]
```diff
@@ -39,6 +39,14 @@ export default Model.extend({
...
   start: attr("django-date"),
 
+  /**
+   * Whether the user is an external employee
+   *
+   * @property {Boolean} isExternal
+   * @public
+   */
+  isExternal: attr("boolean", { defaultValue: false }),
+
```
Comment (anehx) [inline app/models/employment.js:49]: Refactor the whole model to class syntax

File: app/statistics/template.hbs [hunk 9]
```diff
@@ -1,202 +1,209 @@
...
+        data               = data
+        type               = type
+        missingParams      = missingParams
+        ordering           = ordering
+        class              = 'grid-cell'
+        on-ordering-change = (action (mut ordering))
+      }}
+    </div>
+  {{/if}}
 {{/if}}
```
Comment (anehx) [inline app/statistics/template.hbs:209]: Angle bracket syntax, for the whole file..

File: app/statistics/template.hbs [hunk 10]
```diff
@@ -1,202 +1,209 @@
 {{#if (or (media 'isMo') (media 'isXs') (media 'isSm'))}}
   {{no-mobile-message}}
 {{else}}
-  <h1>Statistics</h1>
+  {{#if (cannot "access page")}}
+    <div class="text-center" data-test-access-forbidden>
+      <h1>Access forbidden</h1>
+      <h4>You do not have the permission to access this page</h4>
+    </div>
+  {{else}}
```
Comment (anehx) [inline app/statistics/template.hbs:9]: You should make a wrapper component for this. E.g:

```hbs
{{!-- page-permission.hbs --}}

{{#if (cannot "access page")}}
  <div class="text-center" data-test-access-forbidden>
    <h1>Access forbidden</h1>
    <h4>You do not have the permission to access this page</h4>
  </div>
{{else}}
 {{yield}}
{{/if}}
```
Which you can call then like this:
```hbs
<PagePermission>
  <h1>Statistics</h1>
  ...
</PagePermission>
```

**Review by anehx — CHANGES_REQUESTED** [review]
Start refactoring old code you touch. Otherwise the upgrade to Ember v4 will be a real pain.

**Comment by anehx** [comment]
Also, tests are failing..

File: app/abilities/page.js (hunk 1 above)
Comment (trowik) [reply app/abilities/page.js:4]: Is it compatible with Ember version 3.12?
We had a similar discussion in another PR
https://github.com/adfinis-sygroup/timed-frontend/pull/562#discussion_r657992139

Comment (anehx) [reply app/abilities/page.js:4]: In other words the update was not done in another PR. Nice.

File: app/components/sy-topnav/template.hbs (hunk 5 above)
Comment (anehx) [reply app/components/sy-topnav/template.hbs:37]: Resolved because update didn't happen

File: app/index/template.hbs (hunk 6 above)
Comment (anehx) [reply app/index/template.hbs:54]: Resolved because update didn't happen

File: app/index/template.hbs (hunk 7 above)
Comment (anehx) [reply app/index/template.hbs:93]: Resolved because update didn't happen

File: app/models/employment.js (hunk 8 above)
Comment (anehx) [reply app/models/employment.js:49]: Resolved because update didn't happen

File: app/statistics/template.hbs (hunk 9 above)
Comment (anehx) [reply app/statistics/template.hbs:209]: Resolved because update didn't happen

File: app/abilities/page.js (hunk 1 above)
Comment (anehx) [reply app/abilities/page.js:4]: Resolved because update didn't happen

**Comment by trowik** [comment]
Tests have passed

**Review by anehx — APPROVED** [review]

**Pull request merged**